multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/usage/tracker/RamResourceUsageTracker.java

```diff
@@ -72,14 +73,17 @@ public RamResourceUsageTracker(
       if (WorkspaceStatus.STARTING.equals(activeWorkspace.getStatus())) {
         // starting workspace may not have all machine in runtime
         // it is need to calculate ram from environment config
-        final EnvironmentImpl startingEnvironment =
-            activeWorkspace
-                .getConfig()
-                .getEnvironments()
-                .get(activeWorkspace.getRuntime().getActiveEnv());
-        if (startingEnvironment != null) {
-          currentlyUsedRamMB += environmentRamCalculator.calculate(startingEnvironment);
+        WorkspaceConfigImpl config = activeWorkspace.getConfig();
+
+        if (config != null) {
+          final EnvironmentImpl startingEnvironment =
+              config.getEnvironments().get(activeWorkspace.getRuntime().getActiveEnv());
+          if (startingEnvironment != null) {
+            currentlyUsedRamMB += environmentRamCalculator.calculate(startingEnvironment);
+          }
         }
+        // Estimation of memory for starting workspace with Devfile is not implemented yet
+        // just ignore such
```

Comment:
> With #13202 merged I think we have all the information we can get, haven't we? 

it allows to users optionally define plugin/editor memory. But if user does not do it, then we do not have information about plugin/editor memory.
I believe it should be fixed in scope of [1](https://github.com/eclipse/che/issues/11424)  after merging meta.yaml and che-plugin.yaml  [2](https://github.com/eclipse/che/issues/12908)

> Also, IMHO we should be more vocal here and maybe even log a warning or somesuch, so that this problem is more prominent.

Good point. Will consider adding a warning here.

Reply:
Since it actually does not introduce new bug but adds one more place where it is actual I would like to add logging in a separate PR to avoid discussion of changes which have not strong dependency on introducing devfile as part of workspace https://github.com/sleshchenko/che/commit/60ae656058e25ba626e59eff8dbfdc748ddd7341
Are you OK with that?

Reply:
Will be handled in a separate PR after merging current one